### ISSUE TITLE
suggestion: mathjax in remark42.

### ISSUE THREAD

**Issue by joseph-vidal-rosset** [umputun/remark42#478]
Hi. 
I believe that it would be both nice and very easy to get mathjax in remark42, because it requires only to provide a script in any html page produced via remark42. I do not understand the detail of org-export-head code https://github.com/itf/org-export-head that I use for my blog, but at any html export of a new post that contains a formula that can be translated by mathjax, there is the following script in the html exported page, just before </head>:

```
<!--/*--><![CDATA[/*><!--*/
 function CodeHighlightOn(elem, id)
 {
   var target = document.getElementById(id);
   if(null != target) {
     elem.cacheClassElem = elem.className;
     elem.cacheClassTarget = target.className;
     target.className = "code-highlighted";
     elem.className   = "code-highlighted";
   }
 }
 function CodeHighlightOff(elem, id)
 {
   var target = document.getElementById(id);
   if(elem.cacheClassElem)
     elem.className = elem.cacheClassElem;
   if(elem.cacheClassTarget)
     target.className = elem.cacheClassTarget;
 }
/*]]>*///-->
</script>`
`<script type="text/x-mathjax-config">
    MathJax.Hub.Config({
        displayAlign: "center",
        displayIndent: "0em",
        "HTML-CSS": { scale: 100,
                        linebreaks: { automatic: "false" },
                        webFont: "TeX"
                       },
        SVG: {scale: 100,
              linebreaks: { automatic: "false" },
              font: "TeX"},
        NativeMML: {scale: 100},
        TeX: { equationNumbers: {autoNumber: "AMS"},
               MultLineWidth: "85%",
               TagSide: "right",
               TagIndent: ".8em"
             }
}); </script>`
`<script type="text/javascript"
        src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS_HTML"></script>
```

And formulas in html page are very clean. This system is already use for comments here:
https://math.stackexchange.com/ , see for example: https://math.stackexchange.com/questions/1705236/show-that-a-theory-is-complete-if-and-only-if-b-or-c-is-provable-implies-eith . if I have correctly undestood, this comment  engine is in  "staticman": https://staticman.net/ and it works with GitHub. In my opinion, it would not be difficult at all to provide this script for all messages generated via remark42, or for only those that required it. I am aware that of course that it is certainly not a "popular request", but it would be an additional feature of remark42 that could be appreciated by some of their users.

**Comment by joseph-vidal-rosset** [comment]
Previously, please read "just before `</head>`"

**Comment by paskal** [comment]
My vote is against that because of it would be unused by most but the very few sites and there are existing solutions ([latex2png](https://latex2png.com/), [MathJax convert to image](https://codepen.io/pkra/pen/PZLyQO), [MathML Cloud](https://mathmlcloud.org/), [Online LaTeX Equation Editor](https://www.codecogs.com/latex/eqneditor.php)) which allow you to embed formulas as images.
The only case this wouldn't be enough I can think of is some kind of science- or math-related site where users would share formulas frequently, and yet I would think twice about adding such functionality for everyone for the sake of such a scenario.

**Comment by joseph-vidal-rosset** [comment]
I am surprised, because  I believe that a script to a cdn mathjx server in
the html of remark42 replies would be enough to povide this function.  If I
am right,  it would take one minute to write this script. People can use
mathjax not . It would cost nothing. ..

Le dim. 22 déc. 2019 23:54, Dmitry Verkhoturov <notifications@github.com> a
écrit :

**Comment by paskal** [comment]
I understand that it very trivial to implement and my only concern is increasing complexity of whatever you can put into the comment field and not the software complexity.

I think if you want to add your LaTeX math function in the comment, your expectation today is it's not supported and you need to generate an image with it first. I've never seen a site that supported formulas in the comment except for the one [you mentioned](https://math.stackexchange.com/), and that says something about the demand for this feature. Based on that I see we are not leaving someone with unfulfilled expectations: users on most of the sites don't expect this from the comment system in the first place.

<img width="663" alt="image" src="https://user-images.githubusercontent.com/712534/71379193-b942b680-25ca-11ea-8549-4be99666ec49.png">

On the bottom right there is a text "Styling with Markdown supported". It's up to Remark42 maintainer to decide, but I am strongly against having something like "Styling with Markdown and MathJax is supported" unless I'll see some reasoning for that better than "it's cheap to add" (it's not an argument at all) and "would be used on a single site". If there is a feature that will be turned ON on a single installation only, it should not be implemented in the first place in my opinion.

**Comment by joseph-vidal-rosset** [comment]
There is a misunderstanding. No user have to decide if he wants use or not use mathjax.  Mathjax works immediately when you write a formula in latex code inside $$ or inside \( \) . If you do not  write such a formula you do not know  that mathjax is supported or not. Now, it was only a suggestion.  Best wishes and merry Christmas.

**Comment by joseph-vidal-rosset** [comment]
It seems to me that you do not know how mathjax works for someone that is writing hence a misunderstanding.  The user has not to decide if mathjax must be implemented or not, because  if the mathjax script is in the html page, it will work for any latex code inside $$ or inside antislash and brackets. Now, it was only a suggestion to make Remark42 still more powerful.  If you do not want implement it and if I see that I need it, I will see another solution.  BEST wishes and merry  Christmas.

**Comment by umputun** [comment]
> because if the mathjax script is in the html page, it will work for any latex code inside $$ or inside antislash and brackets

@joseph-vidal-rosset just to clarify - adding a script to the frontend, rendering mathjax (or any other special rendering) is not the solution here, because of remark42 renders everything on the server-side and stores the final html. This final html served directly by the backend, i.e. no special formatting symbols even parsed on the frontend side.

**Comment by joseph-vidal-rosset** [comment]
Thanks for this useful clarification.
Merry  Christmas!
